schemas/core/components/personal-data-query.json

```diff
@@ -0,0 +1,75 @@
+{
+  "$id": "http://maasglobal.com/core/components/personal-data-query.json",
+  "description": "MaaS personal data query parameters schema",
+  "definitions": {
+    "queryParams": {
+      "type": "object",
+      "properties": {
+        "customer[identityId]": {
```

Comment:
I think fact that GET query is flat should not affect JSON representation that we validate.

We had similar situation in `ontrack` and over there @hieuunguyeen used [qs](https://www.npmjs.com/package/qs) to unserialize nested object that came with GET before validation.

That we way we may reuse validation for same nested structures across GET and POST requests.

Reply:
Otrack querystring validation schema is here https://github.com/maasglobal/maas-schemas/blob/develop/schemas/tsp/journey-planner/request.json#L5-L8

Not really performing proper validation

Reply:
I just remember discussing with @hieuunguyeen nesting properties, but didn't look into outcome.

Anyway I think we should just reuse same structures across GET and POST requests, and configure schema only for normal object structures not influenced by limitation of a transport method.

Reply:
Not dismissing your idea, it could work nicely but requires more effort and in that case I would propose postponing this PR and first merging the TSP changes that we need to get for next release and then apply validation on next release

Reply:
I also think that schemas should be validating the raw request coming in, regardless of transportation method.
Defining schema with normal object structure hinting that request is required to go through parser before it could be validated.

Reply:
> I also think that schemas should be validating the raw request coming in,

@hieuunguyeen if we follow this advice, then schema's should validate query string in direct forms  (so e.g. `foo=bar&other=etc` should be passed to schema validate).

JSON schemas are purely about JSON objects, and to comply to that, we _already_ do not validate _raw_ request, but parse query string format into JSON before validation. I don't think it should be problematic to improve serialization/deserialization method we already have on our ends so it can also handle nested object structures.

Big downside of what's proposed here is that it forces us to maintain two schemas for exactly same data (once handled in format as `{ customer: { lastName, firstName } }`, and then as `{ "customer[firstName]" , "customer[lastName]" }`.

Design and maintenance wise it looks quite bad to me.